QModManager/Utility/IOUtilities.cs

```diff
@@ -22,31 +22,36 @@ internal static class IOUtilities
             "steam_shader_cache",
         };
 
-        internal static string GetFolderStructureAsTree(string directory = null)
+        internal static void LogFolderStructureAsTree(string directory = null)
         {
             try
             {
                 directory ??= Environment.CurrentDirectory;
 
-                return GenerateFolderStructure(directory);
+                GenerateFolderStructure(directory);
             }
             catch (Exception e)
             {
                 throw e;
             }
         }
 
-        internal static string GenerateFolderStructure(string directory)
+        internal static void GenerateFolderStructure(string directory)
         {
             var builder = new StringBuilder();
+
             try
             {
                 builder.AppendLine();
                 builder.AppendLine($"+ {new DirectoryInfo(directory).Name}");
 
+                Logger.Info(builder.ToString());
+                builder.Clear();
```

Comment:
Suggestion for your consideration:

Instead of modifying the original method in a way that seems a little hacky (clearing the builder rather than just making a new string when you're not using any of the other features of the builder anyway?), why not just take the result from `string GenerateFolderStructure` as usual and split it to an array by `Environment.NewLine`, and loop through the array to print it?

That, or, if you're going to change this method so that is *only* for *printing* the folder structure rather than for *generating and returning it as a string*, I would rename it to `PrintFolderStructure` instead of just removing the `return`.

If you go with the second option (which is viable), I'd essentially recommend ditching the `StringBuilder` entirely and reworking it to just print each line.

Reply:
The intent was to take the contents of each, top-level folder and print that as a single log.  
That would still include several line breaks within the log.

Thoughts?

Reply:
I see! My apologies I somehow completely missed that in your description.

I like the idea. I would still move toward renaming the method as it is now just a printer rather than generating and returning something.